src/components/diagrams/diagram-common.js

```diff
@@ -0,0 +1,323 @@
+    }, [diagramStates, studyUuid]);
+
+    const openDiagramView = useCallback(
+        (id, type) => {
+            dispatch(openDiagram(id, type));
+        },
+        [dispatch]
+    );
+
+    const togglePinDiagramView = useCallback(
+        (id, type) => {
+            dispatch(togglePinDiagram(id, type));
+        },
+        [dispatch]
+    );
+
+    const minimizeDiagramView = useCallback(
+        (id, type) => {
+            dispatch(minimizeDiagram(id, type));
+        },
+        [dispatch]
+    );
+
+    const closeDiagramView = useCallback(
+        (id, type) => {
+            dispatch(closeDiagram(id, type));
+        },
+        [dispatch]
+    );
+
+    const closeDiagramViews = useCallback(
+        (idsToRemove) => {
+            dispatch(closeDiagrams(idsToRemove));
+        },
+        [dispatch]
+    );
+
+    return [
+        openDiagramView,
+        minimizeDiagramView,
+        togglePinDiagramView,
+        closeDiagramView,
+        closeDiagramViews,
+    ];
+};
+
+export function getNameOrId(value) {
+    return value?.name ?? value?.id;
+}
+
+export function getSubstationNameOrId(value) {
+    return value?.substationName ?? value?.substationId;
+}
+
+export const NoSvg = { svg: null, metadata: null, error: null, svgUrl: null };
+
+// Compute the paper and svg sizes. Returns undefined if the preferred sizes are undefined.
+export const computePaperAndSvgSizesIfReady = (
+    isFullScreenActive,
+    svgType,
+    totalWidth,
+    totalHeight,
+    svgPreferredWidth,
+    svgPreferredHeight,
+    headerPreferredHeight
+) => {
+    if (
+        typeof svgPreferredWidth != 'undefined' &&
+        typeof headerPreferredHeight != 'undefined'
+    ) {
+        let paperWidth, paperHeight, svgWidth, svgHeight;
+        if (isFullScreenActive) {
+            paperWidth = totalWidth;
+            paperHeight = totalHeight;
+            svgWidth = totalWidth - BORDERS;
+            svgHeight = totalHeight - headerPreferredHeight - BORDERS;
+        } else {
+            let tempMaxWidth, tempMaxHeight;
+            if (svgType === SvgType.VOLTAGE_LEVEL) {
+                tempMaxWidth = MAX_WIDTH_VOLTAGE_LEVEL;
+                tempMaxHeight = MAX_HEIGHT_VOLTAGE_LEVEL;
+            } else if (svgType === SvgType.SUBSTATION) {
+                tempMaxWidth = MAX_WIDTH_SUBSTATION;
+                tempMaxHeight = MAX_HEIGHT_SUBSTATION;
+            } else if (svgType === SvgType.NETWORK_AREA_DIAGRAM) {
+                tempMaxWidth = MAX_WIDTH_NETWORK_AREA_DIAGRAM;
+                tempMaxHeight = MAX_HEIGHT_NETWORK_AREA_DIAGRAM;
+            } else {
+                console.error('type inconnu svgType');
+            }
+            svgWidth = Math.min(
+                svgPreferredWidth,
+                svgType !== SvgType.NETWORK_AREA_DIAGRAM
+                    ? totalWidth //- MAP_RIGHT_OFFSET
+                    : totalWidth, // MAP_RIGHT_OFFSET = 120 pour SLD, 0 pour NAD
```

Comment:
why do you have a condition ?

Reply:
This is from old code. It is now removed.